src/main/java/frc/robot/subsystems/drive/Drivetrain.java

```diff
@@ -278,54 +369,56 @@ private Vector limitDirectionSlewRate(Vector spdVec) {
     }
 
     double currentTime = m_timer.get();
-    double elapsedTime = currentTime - m_prevTime;
+    double elapsedTime = currentTime - m_prevSlewRateTime;
 
     double angleDif =
         SwerveUtils.AngleDifference(inputTranslationDir, m_currentTranslationDirRadians);
 
-    m_prevTime = currentTime;
-
     if (angleDif < DriveConfig.MIN_ANGLE_SLEW_RATE) {
       m_currentTranslationDirRadians =
           SwerveUtils.StepTowardsCircular(
               m_currentTranslationDirRadians, inputTranslationDir, directionSlewRate * elapsedTime);
       m_currentTranslationMag = m_magLimiter.calculate(inputTranslationMag);
-      return (new Vector(m_currentTranslationMag, 0)).rot(m_currentTranslationDirRadians);
-    }
-
-    if (angleDif > DriveConfig.MAX_ANGLE_SLEW_RATE) {
-      if (SwerveUtils.approxEqual(
-          m_currentTranslationMag,
-          0)) { // some small number to avoid floating-point errors with equality checking
-        // keep currentTranslationDir unchanged
+      SmartDashboard.putNumber("translation magnitude output", inputTranslationMag);
+    } else if (angleDif > DriveConfig.MAX_ANGLE_SLEW_RATE) {
+      if (m_currentTranslationMag
+          > 1e-4) { // some small number to avoid floating-point errors with equality checking
         m_currentTranslationMag = m_magLimiter.calculate(0.0);
-        return (new Vector(m_currentTranslationMag, 0)).rot(m_currentTranslationDirRadians);
+      } else {
+        m_currentTranslationDirRadians =
+            SwerveUtils.WrapAngle(m_currentTranslationDirRadians + Math.PI);
+        m_currentTranslationMag = m_magLimiter.calculate(inputTranslationMag);
       }
-
+    } else {
       m_currentTranslationDirRadians =
-          SwerveUtils.WrapAngle(m_currentTranslationDirRadians + Math.PI);
-      m_currentTranslationMag = m_magLimiter.calculate(inputTranslationMag);
-      return (new Vector(m_currentTranslationMag, 0)).rot(m_currentTranslationDirRadians);
-    }
+          SwerveUtils.StepTowardsCircular(
+              m_currentTranslationDirRadians, inputTranslationDir, directionSlewRate * elapsedTime);
 
-    m_currentTranslationDirRadians =
-        SwerveUtils.StepTowardsCircular(
-            m_currentTranslationDirRadians, inputTranslationDir, directionSlewRate * elapsedTime);
+      m_currentTranslationMag = m_magLimiter.calculate(0.0);
 
-    m_currentTranslationMag = m_magLimiter.calculate(0.0);
+      m_prevSlewRateTime = currentTime;
+    }
+    spdVec.setX(m_currentTranslationMag);
+    spdVec.setY(0);
+    spdVec.rot(m_currentTranslationDirRadians);
+  }
 
-    return (new Vector(m_currentTranslationMag, 0)).rot(m_currentTranslationDirRadians);
+  /**
+   * checks whether pathplanner paths should be flipped based on the current alliance
+   *
+   * @return whether pathplanner paths should be flipped
+   */
+  private boolean allianceCheck() {
+    var alliance = DriverStation.getAlliance();
```

Comment:
Why?

Reply:
For auto to mirror paths if necessary. It will be used later and doesn't break anything. I don't want to deal with VC ugliness later so can we merge this too.

Reply:
Remove unused code

Reply:
It's used when AutoBuilder is instantiated